kyo-data/shared/src/main/scala/kyo/TypeMap.scala

```diff
@@ -21,7 +21,8 @@ object TypeMap:
                 end while
                 fatal
             end search
-            self.getOrElse(t.erased, search).asInstanceOf[B]
+            if isEmpty && t =:= Tag[Any] then ().asInstanceOf[B]
```

Comment:
Could we have issues with other super types like `AnyRef`, `AnyVal`, `Matchable`?

Reply:
No because those don't conform to the type bound `Any`. I can add a test for it if you want.

`TypeMap.empty.get[AnyRef]` fails to compile.

Reply:
I see, makes sense. I think tests for that can be useful to avoid issues with future changes.